macros/metrics/base/expression.sql

```diff
@@ -40,36 +41,70 @@
     {% do metrics_to_compute.extend(var('re_data:metrics_base')['table']) %}
     {% do metrics_to_compute.extend(metrics.get('table', [])) %}
 
-    {% for metric in metrics_to_compute %}
-        {% set expression = re_data.metrics_base_expression_table(time_filter, metric) %}
-        {% do table_expr.append({ 'expr': expression, 'col_name': '', 'metric': metric}) %}
+    {% for metric_value in metrics_to_compute %}
+        {% set metric_obj = re_data.extract_metric_config(metric_value) %}
+        {% set expression = re_data.metrics_base_expression_table(time_filter, metric_obj['metric'], metric_obj['config']) %}
+        {% do table_expr.append({ 'expr': expression, 'col_name': '', 'metric': metric_obj['metric']}) %}
     {% endfor %}
 
     {{ return (table_expr) }}
 
 {% endmacro %}
 
-{% macro metrics_base_expression_table(time_filter, metric) %}
+{% macro metrics_base_expression_table(time_filter, metric, config) %}
     {% set macro_name = 're_data_metric' + '_' + metric %}
+    {% set project_context = re_data.get_project_context(macro_name, metric) %}
 
-    {% if context['re_data'].get(macro_name) %}
-        {{ context['re_data'][macro_name](time_filter) }}
+    {% if config is not none %}
+        {{ project_context[macro_name](time_filter, config) }}
     {%- else %}
-        {{ context[project_name][macro_name](time_filter) }}
+        {{ project_context[macro_name](time_filter) }}
     {% endif %}
 
 {% endmacro %}
 
 
-{%- macro metrics_base_expression_column(column_name, metric) %}
+{%- macro metrics_base_expression_column(column_name, metric, config) %}
     {% set macro_name = 're_data_metric' + '_' + metric %}
+    {% set project_context = re_data.get_project_context(macro_name, metric) %}
 
-    {% if context['re_data'].get(macro_name) %}
-        {{ context['re_data'][macro_name](column_name) }}
+    {% if config is not none %}
+        {{ project_context[macro_name](column_name, config) }}
     {%- else %}
-        {{ context[project_name][macro_name](column_name) }}
+        {{ project_context[macro_name](column_name) }}
     {% endif %}
 
 {% endmacro %}
 
+{% macro extract_metric_config(metric_value) %}
+
+    {% set config = none %}
+
+    {% if metric_value is mapping %}
+        {% set metric = metric_value.keys() | first %}
+        {% if metric_value[metric] is none %}
+            {{ exceptions.raise_compiler_error("Empty configuration passed for metric: " ~ metric ~ ". If the metric doesn't use a config, please use the column name as a string.") }}
+        {% endif %}
+
+        {% set config = metric_value[metric] %}
+    {%- else %}
+        {% set metric = metric_value %}
+    {% endif %}
+
+    {{ return ({'metric': metric, 'config': config}) }}
+
+{% endmacro %}
+
+{%- macro get_project_context(macro_name, metric) %}
+    {% set macro_name = 're_data_metric' + '_' + metric %}
+
+    {% if context['re_data'].get(macro_name) %}
+        {% set project_context = context['re_data'] %}
```

Comment:
small thing, I think here just could rename a macro to get_metric_macro etc. and already return macro? 
It would be a bit clearer otherwise you still need to access this context from outside of function and name is a bit confusing.

Reply:
👍